crates/net/network/src/session/mod.rs

```diff
@@ -1022,6 +1019,9 @@ async fn authenticate_stream(
                 .ok();
         }
 
+        let (multiplex_stream, their_status) =
+            multiplex_stream.into_eth_satellite_stream(status, fork_filter).await.unwrap();
```

Comment:
I'm also not too sure about this unwrap, if this only applies to the eth cap or if this can also return in an error if the handshake fails

could we convert this into match and return gracefully instead, like

https://github.com/paradigmxyz/reth/blob/4984bc6cc9ff89136f639b6198ad3ba553f3155d/crates/net/network/src/session/mod.rs#L997-L1004

Reply:
good point, done here https://github.com/paradigmxyz/reth/pull/9086/commits/ce008f37983d316d1a50d634ed03a929cca09782